### ISSUE TITLE
⌨ Missing headings on assessment modals

### ISSUE THREAD

**Issue by BrookeCameron** [GCTC-NTGC/gc-digital-talent#14510]
### ⌨ Accessibility and Expected Behaviour

Missing headings inside assessment modals. 

There should be headings for each section within the modal. 

🛡️ reported by Fable tester

### WCAG 2.1 Criteria

1.3.1 Info and Relationships

### 🕵️ Details

Users currently have to navigate through the entirety of the content before they can reach the assessment questions. 

Having section headings inside the modal would allow users to navigate to the next section without having to hear the entire content once they have received the answers they need. 

### What operating systems are you encountering the problem on?

Windows

### What browsers are you encountering the problem on?

Firefox

### What assistive technologies are you encountering the problem on?

NonVisual Desktop Access (NVDA)

### 📸 Screenshot

<img width="696" height="907" alt="Image" src="https://github.com/user-attachments/assets/25807ffe-7534-4a04-b18a-fefcd4430d61" />

### 🙋‍♀️ Proposed Solution

Add headings inside of modal for ease of navigation and to comply with info and relationships WCAG standards
